src/typed-method-types/workflows/triggers/event-data.ts

```diff
@@ -0,0 +1,1301 @@
+    /**
+     * The text in the message.
+     */
+    text: "{{data.text}}",
+    /**
+     * A unique UNIX timestamp in seconds indicating when the threaded message was sent. At runtime this value may not exist if the message in question is not a threaded message!
+     */
+    thread_ts: "{{data.thread_ts}}",
+    /**
+     * A unique identifier for the Slack user who posted the message.
+     */
+    user_id: "{{data.user_id}}",
+  },
+  /**
+   * A message was pinned to a channel.
+   */
+  pin_added: {
+    /**
+     * A unique identifier for the channel where the message was pinned.
+     */
+    channel_id: "{{data.channel_id}}",
+    /**
+     * The channel name for the channel where the message was pinned.
+     */
+    channel_name: "{{data.channel_name}}",
+    /**
+     * The channel type for the channel where the message was pinned. Can be one of "public", "private", "mpdm" or "im".
+     */
+    channel_type: "{{data.channel_type}}", // TODO: validate the various values for this property
+    /**
+     * The event type being invoked. At runtime will always be "slack#/events/pin_added".
+     */
+    event_type: "{{data.event_type}}",
+    /**
+     * A unique UNIX timestamp in seconds indicating when the message being pinned was sent.
+     */
+    message_ts: "{{data.message_ts}}",
+    /**
+     * A unique identifier for the Slack user who pinned the message.
+     */
+    user_id: "{{data.user_id}}",
+  },
+  /**
+   * A message was unpinned from a channel.
+   */
+  pin_removed: {
+    /**
+     * A unique identifier for the channel where the message was unpinned.
+     */
+    channel_id: "{{data.channel_id}}",
+    /**
+     * The channel name for the channel where the message was unpinned.
+     */
+    channel_name: "{{data.channel_name}}",
+    /**
+     * The channel type for the channel where the message was unpinned. Can be one of "public", "private", "mpdm" or "im".
+     */
+    channel_type: "{{data.channel_type}}", // TODO: validate the various values for this property
+    /**
+     * The event type being invoked. At runtime will always be "slack#/events/pin_removed".
+     */
+    event_type: "{{data.event_type}}",
+    /**
+     * A unique UNIX timestamp in seconds indicating when the message being unpinned was sent.
+     */
+    message_ts: "{{data.message_ts}}",
+    /**
+     * A unique identifier for the Slack user who unpinned the message.
+     */
+    user_id: "{{data.user_id}}",
+  },
+  /**
+   * An emoji reaction was added to a message.
+   */
+  reaction_added: {
+    /**
+     * A unique identifier for the channel where the emoji reaction was added to.
+     */
+    channel_id: "{{data.channel_id}}",
+    /**
+     * The event type being invoked. At runtime will always be "slack#/events/reaction_added".
+     */
+    event_type: "{{data.event_type}}",
+    /**
+     * A {@link https://api.slack.com/future/types#message-context Message Context} object representing the message being reacted to.
```

Comment:
👨‍🍳💋 Calling out when context data is of a certain Slack type is a nice touch. We could potentially do that for other Slack types (`interactivity`, `channel_id`, `message_ts`, `user_id`, etc.)